sql/core/src/test/scala/org/apache/spark/sql/DataFrameNaFunctionsSuite.scala

```diff
@@ -460,4 +460,12 @@ class DataFrameNaFunctionsSuite extends QueryTest with SharedSparkSession {
       Row(0, 0L, 0.toShort, 0.toByte, Float.NaN, Double.NaN) ::
       Row(0, 0L, 0.toShort, 0.toByte, Float.NaN, Double.NaN) :: Nil)
   }
+
+  test("SPARK-34417 - test fillMap() for column with a dot in the name") {
+    val na = "n/a"
+    checkAnswer(
+      Seq(("abc", 23L), ("def", 44L), (null, 0L)).toDF("ColWith.Dot", "Col")
+        .na.fill(Map("`ColWith.Dot`" -> na)),
```

Comment:
I think this will fail if you have `Map("Col" -> na)` because of the same reason due to Line 424 in `DataFrameNaFunctions.scala`?

Reply:
No, it does fail. Since column is successfully resolved at ```df.resolve(colName)```, it will not go to 424.
Only for the data frame columns not present in the na-fill-map, it will go to line 424.

Reply:
@imback82 i think your point is that if data frame has a column having dot in the name, but it is not part of the null fill map then it will fail.
Yes, it will fail. It will need fix in the catalyst layer.
`import org.apache.spark.sql.SparkSession

object ColumnNameWithDot {

  def main(args: Array[String]): Unit = {

    val spark = SparkSession.builder.appName("Simple Application")
      .config("spark.master", "local").getOrCreate()

    spark.sparkContext.setLogLevel("OFF")

    import spark.implicits._
    val df = Seq(("abc", 23), ("def", 44), (null, 0)).toDF("ColWith.Dot", "Col.2")
    df.na.fill(Map("`ColWith.Dot`" -> "na"))
      .show()
  }
}`

Reply:
Can you run this test after changing it to `Map("Col" -> na)`? I see the following failure, which should be handled:
```
sbt:spark-sql> testOnly *DataFrameNaFunctionsSuite -- -z "SPARK-34417"
[info] - SPARK-34417 - test fillMap() for column with a dot in the name *** FAILED *** (2 seconds, 274 milliseconds)
[info]   org.apache.spark.sql.AnalysisException: Cannot resolve column name "ColWith.Dot" among (ColWith.Dot, Col); did you mean to quote the `ColWith.Dot` column?
[info]   at org.apache.spark.sql.Dataset.org$apache$spark$sql$Dataset$$resolveException(Dataset.scala:272)
[info]   at org.apache.spark.sql.Dataset.$anonfun$resolve$1(Dataset.scala:263)
...
```

Reply:
Handled the above failure.